language-server/src/main/java/org/sonarlint/languageserver/SonarLintLanguageServer.java

```diff
@@ -171,9 +158,17 @@ private UserSettings() {
 
     private UserSettings(Map<String, Object> params) {
       this.testFilePattern = (String) params.get(TEST_FILE_PATTERN);
-      this.analyzerProperties = (Map) params.getOrDefault(ANALYZER_PROPERTIES, Collections.emptyMap());
+      this.analyzerProperties = getAnalyzerProperties(params);
       this.disableTelemetry = (Boolean) params.getOrDefault(DISABLE_TELEMETRY, false);
     }
+
+    private Map getAnalyzerProperties(Map<String, Object> params) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Make "getAnalyzerProperties" a "static" method. ([squid:S2325](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarlint.core%3Asonarlint-core-parent&branch=sonarts-integration-poc&issues=AWEIztHfZBbiMZMLwMJk&open=AWEIztHfZBbiMZMLwMJk)

[ISSUE_KEY]: AWEIztHfZBbiMZMLwMJk